### PR TITLE
BLUE-276 Restrict first node selection based on the specified node ip and port

### PR DESCRIPTION
**Pull request by jairajdev**

### CODE REVIEW

File: src/Config.ts [hunk 1]
```diff
@@ -184,6 +189,11 @@ let config: Config = {
     txCronSchedule: '*/5 * * * *',
   },
   workerProcessesDebugLog: false,
+  restrictFirstNodeSelection: true,
+  firstNodeInfo: {
+    IP: '127.0.0.1',
+    PORT: 4000,
```
Comment (urnotsam) [inline src/Config.ts]: shouldnt the default port be 9001? 4000 is the port of the archiver

File: src/API.ts [hunk 2]
```diff
@@ -93,16 +94,22 @@ export function registerRoutes(server: FastifyInstance<Server, IncomingMessage,
         reply.send({ success: false, error: 'Signature verification failed' })
         return
       }
+      const ip = signedFirstNodeInfo.nodeInfo.externalIp
```
Comment (kun6fup4nd4) [inline src/API.ts:126]: what if ip in payload will be different from actual ip of the sender ?

File: src/Config.ts (hunk 1 above)
Comment (jairajdev) [reply src/Config.ts]: @urnotsam good catch on that. changed it to 9001 now.

**Review by urnotsam — DISMISSED** [review]

**Review by urnotsam — DISMISSED** [review]

**Review by kun6fup4nd4 — DISMISSED** [review]

**Review by urnotsam — DISMISSED** [review]

**Review by mhanson-github — APPROVED** [review]
I approve my own .patch file changes.

**Comment by chrypnotoad** [comment]
@CodiumAI-Agent /describe

File: src/Config.ts [hunk 3]
```diff
@@ -184,6 +186,8 @@ let config: Config = {
     txCronSchedule: '*/5 * * * *',
   },
   workerProcessesDebugLog: false,
+  restrictFirstNodeSelectionByPublicKey: false,
+  firstNodePublicKey: '',
```
Comment (chrypnotoad) [inline src/Config.ts:190]: Should this be an environment variable? When and how does this get set?

File: src/API.ts (hunk 2 above)
Comment (chrypnotoad) [reply src/API.ts:126]: The IP and Port are not even being checked in here are they? The title of this PR says restricted based off IP and Port, but I only see the public key being checked.
@mhanson-github @jairajdev

**Pull request merged**